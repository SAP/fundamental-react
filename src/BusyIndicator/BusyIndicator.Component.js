import { BusyIndicator } from '../';
import path from 'path';
import React from 'react';
import { ComponentPage, Example } from '../_playground';

export const BusyIndicatorComponent = () => {
    return (
        <ComponentPage
            description={'A **Busy Indicator** informs the user of an ongoing operation.'}
            sourceModulePath={path.join(__dirname, './BusyIndicator')}
            title='Busy Indicator'>
            <Example centered title='Busy Indicator'>
                <BusyIndicator show />
            </Example>

            <Example
                centered
                description={'There are 3 sizes for Busy Indicator: s, m & l'}
                title='Busy Indicator Sizes'>
                <BusyIndicator show size='s' />
                <BusyIndicator show />
                <BusyIndicator show size='l' />
            </Example>
        </ComponentPage>
    );
};
