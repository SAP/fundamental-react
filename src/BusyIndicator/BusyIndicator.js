import { BUSY_INDICATOR_SIZES } from '../utils/constants';
import classnamesBind from 'classnames/bind';
import CustomPropTypes from '../utils/CustomPropTypes/CustomPropTypes';
import PropTypes from 'prop-types';
import React from 'react';
import withStyles from '../utils/withStyles';
import styles from 'fundamental-styles/dist/busy-indicator.css';

const classnames = classnamesBind.bind(styles);

/** A **Busy Indicator** informs the user of an ongoing operation. */
const BusyIndicator = React.forwardRef(({
    className,
    cssNamespace,
    show,
    size,
    localizedText,
    ...props
}, ref) => {

    const busyIndicatorClasses = classnames(
        {
            [`${cssNamespace}-busy-indicator--${size}`]: size === 'm' || size === 'l',
            [`${cssNamespace}-busy-indicator`]: size === 's'
        },
        className
    );

    if (!show)
        return null;

    return (
        <div
            {...props}
            aria-hidden='false'
            aria-label={localizedText.loading}
            className={busyIndicatorClasses}
            ref={ref} >
            <div className={classnames(`${cssNamespace}-busy-indicator--circle-0`)} />
            <div className={classnames(`${cssNamespace}-busy-indicator--circle-1`)} />
            <div className={classnames(`${cssNamespace}-busy-indicator--circle-2`)} />
        </div>
    );
});

BusyIndicator.displayName = 'BusyIndicator';

BusyIndicator.propTypes = {
    /** CSS class(es) to add to the element */
    className: PropTypes.string,
    /** Localized text to be updated based on location/language */
    localizedText: CustomPropTypes.i18n({
        /** aria-label for Busy Indicator component */
        loading: PropTypes.string
    }),
    /** Set to **true** to make Busy Indicator visible */
    show: PropTypes.bool,
    /** Size of the component: 's', 'm' or 'l' */
    size: PropTypes.oneOf(BUSY_INDICATOR_SIZES)
};

BusyIndicator.defaultProps = {
    localizedText: {
        loading: 'Loading'
    },
    size: 'm',
    show: false
};

export default withStyles(BusyIndicator);
