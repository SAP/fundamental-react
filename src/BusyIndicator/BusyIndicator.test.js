import BusyIndicator from './BusyIndicator';
import { mount } from 'enzyme';
import React from 'react';

describe('<BusyIndicator />', () => {
    describe('Prop spreading', () => {
        test('should allow props to be spread to the Busy Indicator component', () => {
            const element = mount(<BusyIndicator data-sample='Sample' show />);

            expect(
                element.getDOMNode().attributes['data-sample'].value
            ).toBe('Sample');
        });
    });
    test('forwards the ref', () => {
        let ref;
        class Test extends React.Component {
            constructor(props) {
                super(props);
                ref = React.createRef();
            }
            render = () => <BusyIndicator ref={ref} show />;
        }
        mount(<Test />);
        expect(ref.current.tagName).toEqual('DIV');
        expect(ref.current.className).toEqual('fd-busy-indicator--m');
    });
});
