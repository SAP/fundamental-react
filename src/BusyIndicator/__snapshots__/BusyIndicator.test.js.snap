// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BusyIndicator /> create busy indicators 1`] = `
<div
  aria-hidden="false"
  aria-label="Loading"
  className="fd-busy-indicator--m"
>
  <div
    className="fd-busy-indicator--circle-0"
  />
  <div
    className="fd-busy-indicator--circle-1"
  />
  <div
    className="fd-busy-indicator--circle-2"
  />
</div>
`;

exports[`<BusyIndicator /> create busy indicators 2`] = `
<div
  aria-hidden="false"
  aria-label="Loading"
  className="fd-busy-indicator"
>
  <div
    className="fd-busy-indicator--circle-0"
  />
  <div
    className="fd-busy-indicator--circle-1"
  />
  <div
    className="fd-busy-indicator--circle-2"
  />
</div>
`;

exports[`<BusyIndicator /> create busy indicators 3`] = `
<div
  aria-hidden="false"
  aria-label="Loading"
  className="fd-busy-indicator--l"
>
  <div
    className="fd-busy-indicator--circle-0"
  />
  <div
    className="fd-busy-indicator--circle-1"
  />
  <div
    className="fd-busy-indicator--circle-2"
  />
</div>
`;

exports[`<BusyIndicator /> create busy indicators 4`] = `null`;

exports[`<BusyIndicator /> hide busy indicator 1`] = `null`;
